configure.ac

```diff
@@ -250,10 +250,18 @@ DIST_PACKAGING([debian ubuntu],[DEBIAN_PACKAGE],[DEBUILD],[debuild])
 DIST_PACKAGING([osx],[OSX_PACKAGE],[PKGBUILD],[pkgbuild])
 DIST_PACKAGING([bsd],[BSD_PACKAGE],[PKGBUILD],[pkgbuild])
 
+dnl ============================libffi=========================================
+PKG_PROG_PKG_CONFIG()
```

Comment:
Can we just use 
```
AC_CHECK_LIB(ffi, ffi_call,,
    [AC_MSG_ERROR([required library ffi missing])])
AC_CHECK_HEADER(ffi.h,,[AC_MSG_ERROR([required headers ffi.h missing.])])
```
I don't fully understand the debates about PKG_CHECK_MODULES, but it will at least help compilation on freeBSD